lib/cobra_commander/component_tree/combined_component_tree.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require "oj"
+
+module CobraCommander
+  class ComponentTree
+    # Represents a tree of all component dependencies
+    class CombinedComponentTree
+      def initialize(name, path, ancestry = Set.new)
+        @name = name
+        @root_path = path
+        @ancestry = ancestry
+      end
+
+      def to_h
+        {
+          name: @name,
+          path: @root_path,
+          ancestry: @ancestry,
+          dependencies: dependencies.map(&method(:dep_representation)),
+        }
+      end
+
+      def node_tree
+        @node_tree ||= NodeComponentTree.new(UMBRELLA_APP_NAME, @root_path).to_h
+      end
+
+      def gem_tree
+        @gem_tree ||= GemComponentTree.new(UMBRELLA_APP_NAME, @root_path).to_h
+      end
```

Comment:
These are essentially useless now. I can remove these methods, these classes, and move the "Combined" logic directly into `ComponentTree`.  

However, it there is value in having a command to generate ONLY the gem dependencies or node dependencies, I can use this methods/model still.  Thoughts?

Reply:
I don't think there's value there. We do need to identify the type of the component in the output though.